internal/appsec/waf.go

```diff
@@ -73,11 +73,15 @@ func registerWAF(rules []byte, timeout time.Duration, limiter Limiter, obfCfg *O
 	var unregisterHTTP, unregisterGRPC dyngo.UnregisterFunc
 	if len(httpAddresses) > 0 {
 		log.Debug("appsec: registering http waf listening to addresses %v", httpAddresses)
-		unregisterHTTP = dyngo.Register(newHTTPWAFEventListener(waf, httpAddresses, timeout, limiter))
+		unregisterHTTP = dyngo.Register(newHTTPWAFEventListener(waf, httpAddresses, a.cfg.wafTimeout, a.limiter))
 	}
 	if len(grpcAddresses) > 0 {
 		log.Debug("appsec: registering grpc waf listening to addresses %v", grpcAddresses)
-		unregisterGRPC = dyngo.Register(newGRPCWAFEventListener(waf, grpcAddresses, timeout, limiter))
+		unregisterGRPC = dyngo.Register(newGRPCWAFEventListener(waf, grpcAddresses, a.cfg.wafTimeout, a.limiter))
+	}
+
+	if err := a.enableRCBlocking(&wafHandleWrapper{waf}); err != nil {
```

Comment:
nit (because not in the hotpath ;-p): small wrappers like this don't need to be passed by address

Reply:
Removed the ref.